src/app/connections/list-toolbar/list-toolbar.component.ts

```diff
@@ -60,7 +62,21 @@ export class ConnectionsListToolbarComponent<T> implements OnInit, OnDestroy {
     } as ToolbarConfig;
 
     this.subscription = this.items
-      .do(connections => this._allItems = connections)
+      .do(items => this._allItems = items)
+      .do(items => {
+        if (_.some(items, item => _.isArray(item['tags']))) {
+          if (!_.find(filterConfig.fields, { id: 'tag' })) {
+            filterConfig.fields.push({
+              id          : 'tag',
+              title       : 'Tag',
+              placeholder : 'Filter by tag...',
+              type        : 'typeahead',
+            });
+          }
+        } else {
+          _.remove(filterConfig.fields, { id: 'tag' });
```

Comment:
`filterConfigFields.slice()` probably?  Can get the index from `find` if I recall correctly.